Bottom: 3be62f5e83ce5f7af11fbd34dd1b502768d85e3e
Top:    899ed3be1de41908bc435ccd73cc6f704b41bf4f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 23:52:14 +0000

Refresh of attempt-at-re-writing-drawing

---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index f5c6e46..19c4708 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -676,15 +676,15 @@ gerber_do_export (HID_Attr_Val * options)
   region.X2 = PCB->MaxWidth;
   region.Y2 = PCB->MaxHeight;
 
-#warning NULL gc
-  common_set_clip_box (NULL, &region);
-
   pagecount = 1;
   resetApertures ();
 
   lastgroup = -1;
   layer_list_idx = 0;
   finding_apertures = 1;
+
+  common_set_clip_box (&gerber_graphics, &region);
+
   hid_expose_callback (&gerber_graphics, 0);
 
   layer_list_idx = 0;
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 2bfa077..c860c5a 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -826,10 +826,7 @@ Benchmark (int argc, char **argv, Coord x, Coord y)
   region.X2 = PCB->MaxWidth;
   region.Y2 = PCB->MaxHeight;
 
-#warning NULL gc
-  gc = NULL;
-
-  common_set_clip_box (gc, &region);
+  common_set_clip_box (&lesstif_graphics, &region);
 
   pixmap = window;
   XSync (display, 0);
@@ -2504,9 +2501,6 @@ idle_proc (XtPointer dummy)
 {
   hidGC gc;
 
-#warning NULL gc
-  gc = NULL;
-
   if (need_redraw)
     {
       int mx, my;
@@ -2590,7 +2584,7 @@ idle_proc (XtPointer dummy)
 	    }
 	}
 
-      common_set_clip_box (gc, &region);
+      common_set_clip_box (&lesstif_graphics, &region);
 
       DrawBackgroundImage();
       hid_expose_callback (&lesstif_graphics, 0);
@@ -3817,12 +3811,9 @@ pinout_callback (Widget da, PinoutData * pd,
   region.X2 = PCB->MaxWidth;
   region.Y2 = PCB->MaxHeight;
 
-#warning NULL gc
-  gc = NULL;
-
-  common_set_clip_box (gc, &region);
-
   XFillRectangle (display, pixmap, bg_gc, 0, 0, pd->v_width, pd->v_height);
+
+  common_set_clip_box (&lesstif_graphics, &region);
   hid_expose_callback (&lesstif_graphics, pd->item);
 
   pinout = 0;
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 2f8a497..55e88bd 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -646,13 +646,11 @@ nelma_start_png_export()
 	region.X2 = PCB->MaxWidth;
 	region.Y2 = PCB->MaxHeight;
 
-#warning NULL gc
-  common_set_clip_box (NULL, &region);
-
 	linewidth = -1;
 	lastbrush = (gdImagePtr)((void *) -1);
 
-	hid_expose_callback(&nelma_graphics, 0);
+	common_set_clip_box (&nelma_graphics, &region);
+	hid_expose_callback (&nelma_graphics, 0);
 }
 
 static void
